testbed/lib/addons/karpenter.ts

```diff
@@ -0,0 +1,94 @@
+import { Construct } from 'constructs';
+import { aws_iam as iam,Duration, CfnResource, StackProps } from 'aws-cdk-lib';
+import { aws_eks as eks } from 'aws-cdk-lib';
+
+export interface KarpenterProps extends StackProps {
+    cluster: eks.Cluster
+    namespace: string
+    version: string
+    nodeRoleName: string
+}
+
+export class Karpenter extends Construct {
+    constructor(scope: Construct, id: string, props: KarpenterProps) {
+        super(scope, id)
+        const ns = props.cluster.addManifest('karpenter-namespace', {
+            apiVersion: 'v1',
+            kind: 'Namespace',
+            metadata: {
+                name: props.namespace
+            }
+        })
+
+        // Controller Role
+        const sa = props.cluster.addServiceAccount('karpenter-controller-sa', {
+            name: "karpenter",
+            namespace: props.namespace
+        })
+        sa.node.addDependency(ns)
+        sa.role.attachInlinePolicy(new iam.Policy(this, 'karpenter-controller-policy', {
```

Comment:
If we hardcode this policy, we should probably hardcode the karpenter version inside this module, since these may drift.